packages/core-mobile/app/seedless/services/SeedlessSession.ts

```diff
@@ -302,31 +268,35 @@ class SeedlessSessionManager {
     cubeSignerResponse: CubeSignerResponse<T>,
     code: string
   ): Promise<Result<CubeSignerResponse<T>, TotpErrors>> {
-    const signerSession = await this.getSignerSession()
-    const response = await cubeSignerResponse.approveTotp(signerSession, code)
+    const signerClient = await this.getSignerClient()
+    const response = await cubeSignerResponse.totpApprove(signerClient, code)
     return { success: true, value: response }
   }
 
   /**
-   * Returns a CubeSigner instance
+   * Returns a CubeSignerClient instance
    */
-  private async getCubeSignerClient(): Promise<CubeSignerClient> {
-    const sessionManager = await this.getSessionManager()
-    return new CubeSignerClient(sessionManager, SEEDLESS_ORG_ID)
-  }
+  async getSignerClient(): Promise<CubeSignerClient> {
+    if (this.signerClient) {
+      return this.signerClient
+    }
 
-  /**
-   * Returns a session manager that can be used to retrieve session data.
-   */
-  private async getSessionManager(): Promise<SignerSessionManager> {
-    return await SignerSessionManager.loadFromStorage(this.sessionStorage)
-  }
+    if (!this.signerClientPromise) {
+      this.signerClientPromise = (async () => {
+        const client = await CubeSignerClient.create(this.sessionManager)
 
-  /**
-   * Returns a signer session
-   */
-  async getSignerSession(): Promise<SignerSession> {
-    return new SignerSession(await this.getSessionManager())
+        if (this.onSessionExpired) {
+          client.addEventListener('session-expired', this.onSessionExpired)
+        }
+
+        this.signerClient = client
+        this.signerClientPromise = null // Reset the promise lock
+        return client
+      })()
+    }
+
+    // return a promise to prevent multiple client creation
+    return this.signerClientPromise
```

Comment:
I noticed we are accessing this method getSignerSession() at the same time in some places. just doing a promise lock so that we only create 1 client.